### PR TITLE
pipeline: check inputs in post-build-status-update

### PR DESCRIPTION
**Pull request by gibfahn**
Make sure PR is a number, and the `POST_STATUS_TO_PR` is set.

Refs: https://github.com/nodejs/build/pull/973#issuecomment-342130374

### CODE REVIEW

File: jenkins/pipelines/post-build-status-update.jenkinsfile [hunk 1]
```diff
@@ -29,6 +29,14 @@ pipeline {
 }
 
 def sendBuildStatus(repo, identifier, status, url, commit, ref) {
+
+    // Fail quietly if upstream job shouldn't be posting to GitHub.
+    if (!PR.isInteger() || POST_STATUS_TO_PR != 'true') {
```
Comment (maclover7) [inline jenkins/pipelines/post-build-status-update.jenkinsfile]: - Should `PR.isInteger()` change to checking if `ref` includes `refs/pull`? We never explicitly send `PR` to `post-build-status-update`

- `POST_STATUS_TO_PR` will need to be a new string parameter and variable passed to this function

Comment (gibfahn) [reply jenkins/pipelines/post-build-status-update.jenkinsfile]: My bad on the first.

For the second, are parameters not passed down as environment variables in pipelines? If not that's a pain.

Comment (maclover7) [reply jenkins/pipelines/post-build-status-update.jenkinsfile]: So yes, the parameters are passed down as environment variables, but there are two annoying thing at play here -- 1) `sendBuildStatus` is a separate function, so we have to pass all parameters to it explicitly 2) `POST_STATUS_TO_PR` also needs to be defined as a parameter for the pipeline

Comment (gibfahn) [reply jenkins/pipelines/post-build-status-update.jenkinsfile]: How about doing this (what I just pushed)?

File: jenkins/pipelines/post-build-status-update.jenkinsfile [hunk 2]
```diff
@@ -10,28 +10,41 @@ pipeline {
...
 
-def sendBuildStatus(repo, identifier, status, url, commit, ref) {
+def sendBuildStatus(params) {
+    // Fail loudly if upstream job is configured wrong.
+    if (params.any{ it.value == '' }) {
+        error('All parameter fields are required.')
+    }
+
+    // Fail quietly if upstream job shouldn't be posting to GitHub.
+    if (params.REF.contains('refs/pull/') || params.POST_STATUS_TO_PR != 'true') {
```
Comment (refack) [inline jenkins/pipelines/post-build-status-update.jenkinsfile]: `!`:
`!params.REF.contains('refs/pull/')`

**Comment by refack** [comment]
Sidebar: why do we keep the `POST_STATUS_TO_PR` arg? Why not try every time?

**Comment by maclover7** [comment]
ping @gibfahn

**Comment by maclover7** [comment]
ping @gibfahn

**Comment by sam-github** [comment]
Stale pipeline PR related to the pipeline experiment.

**Pull request closed without merging**